### PR TITLE
Documentation changes

### PR DESCRIPTION
**Pull request by TanLeYang**
To view the changes:
https://tanleyang.github.io/tp/UserGuide.html

### DG
- Added more manual test cases to cover at least the basics all of our features as per module requirement

### UG
- Fix some md errors
- Change the introduction to be less informal

### Product website
- Got the bookmark icon to show up properly on the product website

### CODE REVIEW

**Review by pennhanlee — APPROVED** [review]
LGTM

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -1130,3 +1214,45 @@ testers are expected to do more *exploratory* testing.
...
+additions to support the suggestions feature which was not present in AB3.
+
+### Logic
+As the complexity of `Model` increased, the complexity of `Logic` would also have to increase. _bookmark_'s `Logic` supports
+many more Commands compared to AB3, including an improved `find` Command, `sort` Command, `note` Command and many more.
+
+Significant effort was put in to implement new logic for _bookmark_ as the `Logic` of _bookmark_ is responsible for
+a lot more compared to AB3. For example, a new `ViewTypeManager` class was created to keep track of what `ViewType` to display
+a given state of the application with so that the `undo` and `redo` commands can work well. We also put significant
+effort into sanitising and validating user input as well as increasing the testing rigour to ensure that _bookmark_'s parsers
```
Comment (angrybunny123) [inline docs/DeveloperGuide.md:1240]: Maybe we can remove the word "sanitising"

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -1130,3 +1214,45 @@ testers are expected to do more *exploratory* testing.
...
+can handle unexpected inputs smoothly.  
+
+### Storage
+The `Storage` was extended to support more options for user preferences such as sorting preference. Furthermore, the 
+more complex attributes of each `Book` mentioned earlier have to be stored into the JSON file in an appropriate format.
+We spent quite a while discussing and deciding on the best way to represent each new attribute in the JSON file.
+
+### Ui
+A lot of effort was put into the `Ui`. Our team put in quite a bit of effort in ensuring that our interface was pleasing
```
Comment (angrybunny123) [inline docs/DeveloperGuide.md]: I realise that you use 'quite a bit' a lot, maybe for some variation here, you can say "Our team put in a lot of effort to ensure that our interface......."

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -1130,3 +1214,45 @@ testers are expected to do more *exploratory* testing.
...
+to the eye and user-friendly. We also introduced a "view switching" feature which allows users to switch between different
+pages which made the Ui a lot more interactive compared to the single page view of AB3's Ui. 
+
+### Overall
+As a whole, the process was not an easy one for the _bookmark_ team. We started off with plans for many features that we
+eventually decided were not suitable. Hence, in the first iterations, we spent quite a bit of time coming up with new
```
Comment (angrybunny123) [inline docs/DeveloperGuide.md]: Similar to the above point, maybe you can use "we spent a considerable amount of time to come up with new...."

File: docs/UserGuide.md [hunk 4]
```diff
@@ -3,13 +3,14 @@ layout: page
 title: User Guide
 ---
 
-_bookmark_ is a desktop application to manage your indulgent bookworm habits! Use _bookmark_ to track your book reading progress 
-or pen down thoughts of mindboggling fan theories about your favourite characters!
+_bookmark_ is a desktop application for readers to manage their reading progress.
 
-_bookmark_ allows you to add and update your beloved books, track your reading progress using our bookmark and even add reading goals and notes to your chosen books!
+_bookmark_ allows you to easily track your **reading progress** so you'll never need to use a physical bookmark or
+lose track of where you stopped reading again. With many interesting features like **note taking** and **goal setting**,
```
Comment (angrybunny123) [inline docs/UserGuide.md]: Maybe we can do "so *that* you'll never need to use a physical bookmark or lose track of where you stopped reading *ever* again"

File: docs/UserGuide.md [hunk 5]
```diff
@@ -3,13 +3,14 @@ layout: page
...
+_bookmark_ will be your faithful reading companion, helping to make reading even more interactive and exciting.
 
-Guess what? _bookmark_ also uses the special Command Line Interface style so if you love to type, you're in luck! So 
-get cracking and start saving your best reads and ideas into _bookmark_, Type away!~
+This user guide introduces what you can do with _bookmark_ and how it can benefit you. With this guide,
+you will be ready to use _bookmark_ to enhance your reading experience!
```
Comment (angrybunny123) [inline docs/UserGuide.md]: Maybe we can do "This user guide introduces you to the many features of _bookmark_ and how you can benefit from them. With this guide, you will be ready to use _bookmark_ to enhance your reading experience.  [remove the exclamation mark]

**Review by pennhanlee — APPROVED** [review]
LGTM!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

File: docs/UserGuide.md [hunk 6]
```diff
@@ -3,13 +3,14 @@ layout: page
...
+_bookmark_ is a desktop application for readers to manage their reading progress.
 
-_bookmark_ allows you to add and update your beloved books, track your reading progress using our bookmark and even add reading goals and notes to your chosen books!
+_bookmark_ allows you to easily track your **reading progress** so that you'll never need to use a physical bookmark or
+lose track of where you stopped reading ever again. With many interesting features like **note taking** and **goal setting**,
+_bookmark_ will be your faithful reading companion, helping to make reading even more interactive and exciting.
 
-Guess what? _bookmark_ also uses the special Command Line Interface style so if you love to type, you're in luck! So 
-get cracking and start saving your best reads and ideas into _bookmark_, Type away!~
+This user guide introduces what you to the many features of _bookmark_ and how you can benefit from them. With this guide,
```
Comment (angrybunny123) [inline docs/UserGuide.md]: I think there is a small typo here, can remove the word 'what'

**Review by angrybunny123 — COMMENTED** [review]
LGTM other than the small typo I pointed out

**Pull request merged**